### PR TITLE
(feat) Adding Chaos Engineering

### PR DESCRIPTION
**Pull request by lanegoolsby**
This is a direct copy of https://github.com/natenho/Mockaco/pull/114, I just addressed the PR feedback. I have a need for this feature and that other PR stalled. :)

### CODE REVIEW

**Comment by lanegoolsby** [comment]
Hey @natenho, mind taking a look at this PR please?

File: src/Mockaco.AspNetCore/Chaos/Strategies/ChaosStrategyException.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Mockaco.Chaos.Strategies;
+
+internal class ChaosStrategyException : IChaosStrategy
+{
+    public Task Response(HttpResponse httpResponse)
+    {
+        httpResponse.StatusCode = (int)HttpStatusCode.InternalServerError;
+
+        var bodyBytes = Encoding.UTF8.GetBytes($"Error {httpResponse.StatusCode}: {HttpStatusCode.ServiceUnavailable}");
```
Comment (natenho) [inline src/Mockaco.AspNetCore/Chaos/Strategies/ChaosStrategyException.cs:13]: The status code in the string is ServiceUnavailable, I suppose you mean InternalServerError, right?

File: src/Mockaco.AspNetCore/DependencyInjection/MockacoServiceCollection.cs [hunk 2]
```diff
@@ -81,5 +85,13 @@ private static IServiceCollection AddInternalServices(this IServiceCollection se
                 .AddTransient<ITemplateTransformer, TemplateTransformer>()
 
                 .AddTemplatesGenerating();
+        
+        private static IServiceCollection AddChaosServices(this IServiceCollection services) =>
+            services
+                .AddSingleton<IChaosStrategy, ChaosStrategyBehavior>()
```
Comment (natenho) [inline src/Mockaco.AspNetCore/DependencyInjection/MockacoServiceCollection.cs:91]: Do we really need singletons here? I would make any dependency as transient by default if I dont need a singleton..it is much easier to deal with concurrent requests in transient objects if we need some context inside the behavior

**Review by natenho — CHANGES_REQUESTED** [review]
@lanegoolsby there you go! The code is very good, we can merge it asap if you could please review the pr comments! :)

**Comment by natenho** [comment]
@lanegoolsby  I took the liberty of making the changes to avoid taking even more time to do the merge. Thank you for your contribution!

**Review by natenho — APPROVED** [review]

**Comment by lanegoolsby** [comment]
Thank you!!

**Pull request merged**